### PR TITLE
C1f2 crop testing

### PR DESCRIPTION
**Pull request by krb78**
SunnyDene (c1f2) crop and soil jsons

### Context
This enables the cropped acres of c1f2 to be simulated in RuFaS. 

The change log doesn't need to be updated. 

### What
The same crop rotation exists on all fields of this farm, but to capture fields in each part of the rotation, there are 3 "fields" simulated: alfalfa, corn/triticale double crop, and wheat. No fertilizer is applied to the fields, only manure. 

Hopefully, you'll find that this simulation runs and plants/harvests crops. Fair warning, it takes A WHILE.

### CODE REVIEW

**Review by jadamchick — APPROVED** [review]
looks good

File: input/data/manure_schedule/c1f2_alfalfa_2020_manure_schedule.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+    "years": [2000, 2000, 2000, 2001, 2001, 2001, 2002, 2002, 2002, 2003, 2003, 2004, 2004, 2005, 2005, 2006],
+    "days": [155, 277, 420, 155, 277, 420, 155, 277, 420, 120, 274, 120, 274, 120, 274 , 60],
```
Comment (Isgotkowitz) [inline input/data/manure_schedule/c1f2_alfalfa_2020_manure_schedule.json]: There is not a 420th day of the year.

File: input/data/manure_schedule/c1f2_corn_2020_manure_schedule.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{
+    "years": [2001, 2001, 2001, 2002, 2002, 2002, 2003, 2003, 2003, 2004, 2004, 2005, 2005, 2006, 2006, 2007],
+    "days": [155, 277, 420, 155, 277, 420, 155, 277, 420, 120, 274, 120, 274, 120, 274 , 60],
```
Comment (Isgotkowitz) [inline input/data/manure_schedule/c1f2_corn_2020_manure_schedule.json]: Same as above

File: input/data/manure_schedule/c1f2_wheat_2020_manure_schedule.json [hunk 3]
```diff
@@ -0,0 +1,14 @@
+{
+    "years": [2000, 2000, 2000, 2001, 2001, 2001, 2002, 2002, 2002, 2003, 2003, 2004, 2004, 2005, 2005, 2006],
+    "days": [155, 277, 420, 155, 277, 420, 155, 277, 420, 120, 274, 120, 274, 120, 274 , 60],
```
Comment (Isgotkowitz) [inline input/data/manure_schedule/c1f2_wheat_2020_manure_schedule.json]: Same as above

File: input/metadata/c1f2_wcrops_metadata.json [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+      "type": "json",
+      "properties": "manure_schedule_properties"
+    },
+  "Alfalfa 2020 Tillage schedule": {
+      "title": "Corn 2020 Tillage schedule.",
+      "description": "Schedule of tillage applications for a field.",
+      "path": "input/data/tillage_schedule/c1f2_alfalfa_2020_tillage_schedule.json",
+      "type": "json",
+      "properties": "tillage_schedule_properties"
+    },
```
Comment (Isgotkowitz) [inline input/metadata/c1f2_wcrops_metadata.json]: All of these inputs are outside of the "files" object, so RuFaS isn't seeing them at all.

File: input/data/field/c1f2_alfalfa_2020_field.json [hunk 5]
```diff
@@ -0,0 +1,15 @@
+
+{
+  "soil_specification": "soil",
+  "crop_specification": "Alfalfa 2020 Crop data",
+  "manure_management_specification": "Alfalfa 2020 Manure schedule",
+  "tillage_management_specification": "Alfalfa 2020 Tillage schedule",
+  "field_size": 540,
```
Comment (Isgotkowitz) [inline input/data/field/c1f2_alfalfa_2020_field.json:8]:
```suggestion
  "fertilizer_management_specification": "no_fert",
  "field_size": 540,
```
Even if there is not going to be any fertilizer applied, that still has to be specified. The above line can be added to each new field input file, and the following input added to the "files" section of the metadata:
```json
    "no_fert": {
      "title": "Fertilizer schedule.",
      "description": "No fertilizer is scheduled to be used.",
      "path": "input/data/fertilizer_schedule/no_fertilizer.json",
      "type": "json",
      "properties": "fertilizer_schedule_properties"
    },
```

File: input/data/field/c1f2_corn_2020_field.json
Comment (Isgotkowitz) [inline input/data/field/c1f2_corn_2020_field.json:1]: Same as above

File: input/data/field/c1f2_wheat_2020_field.json
Comment (Isgotkowitz) [inline input/data/field/c1f2_wheat_2020_field.json:1]: Same as above

**Review by Isgotkowitz — CHANGES_REQUESTED** [review]
The metadata doesn't include the new field files in the "files" section, so the Crop and Soil module isn't simulating anything right now. After that is corrected, the field input files need to add a specification for the fertilizer applications, even though none is actually applied. There is a more detailed comment on this below.

File: input/data/manure_schedule/c1f2_alfalfa_2020_manure_schedule.json (hunk 1 above)
Comment (krb78) [reply input/data/manure_schedule/c1f2_alfalfa_2020_manure_schedule.json]: wait, what? I thought that's what leap year was about?!

File: input/data/crop/c1f2_corn_2020_rotation.json [hunk 6]
```diff
@@ -0,0 +1,63 @@
+{
+    "crop_schedules": [
+        {
+            "crop_species": "alfalfa_hay",
+            "planting_years": [2000],
+            "pattern_repeat": 2,
+            "planting_skip": 6,
+            "harvesting_skip": 6,
+            "planting_days": [182],
+            "harvest_years": [2001, 2001, 2001, 2001, 2001, 2001, 2002, 2002, 2002, 2002, 2000, 2002, 2003, 2003, 2003, 2003, 2003, 2003, 2004],
```
Comment (Isgotkowitz) [inline input/data/crop/c1f2_corn_2020_rotation.json]: The harvest years need to be in ascending order

**Review by Isgotkowitz — CHANGES_REQUESTED** [review]

File: input/data/crop/c1f2_alfalfa_2020_rotation.json [hunk 7]
```diff
@@ -0,0 +1,63 @@
+{
+    "crop_schedules": [
+        {
+            "crop_species": "alfalfa_hay",
+            "planting_years": [1997],
+            "pattern_repeat": 3,
+            "planting_skip": 6,
+            "harvesting_skip": 6,
```
Comment (Isgotkowitz) [inline input/data/crop/c1f2_alfalfa_2020_rotation.json]:
```suggestion
            "harvesting_skip": 3,
```
I think this should only be three. Reason being that the final harvest is in 2001, and harvests should start again in 2005 (after it is planted the year before) - so skip 2002, 2003, 2004.

File: input/data/crop/c1f2_corn_2020_rotation.json [hunk 8]
```diff
@@ -0,0 +1,63 @@
+{
+    "crop_schedules": [
+        {
+            "crop_species": "alfalfa_hay",
+            "planting_years": [2000],
+            "pattern_repeat": 2,
+            "planting_skip": 6,
+            "harvesting_skip": 6,
```
Comment (Isgotkowitz) [inline input/data/crop/c1f2_corn_2020_rotation.json]: Same as above

File: input/data/crop/c1f2_wheat_2020_rotation.json [hunk 9]
```diff
@@ -0,0 +1,63 @@
+{
+    "crop_schedules": [
+        {
+            "crop_species": "alfalfa_hay",
+            "planting_years": [1999],
+            "pattern_repeat": 3,
+            "planting_skip": 6,
+            "harvesting_skip": 6,
```
Comment (Isgotkowitz) [inline input/data/crop/c1f2_wheat_2020_rotation.json]: Same as above

**Review by Isgotkowitz — APPROVED** [review]
LGTM, nice job Kaitlyn. I ran this sim and it plants and harvests everything correctly (I believe). Before merging, I think the following files may need to be reverted to their original state: `scenario_manager.py`, `default_animal.json`, and `c1f2_config.json`. Also the weather in the C1F2 metadata might need to be changed back to the C1F2 weather file.

When

**Pull request merged**